### PR TITLE
refactor tests into single executable

### PR DESCRIPTION
**Pull request by headupinclouds**
Refactor tests into single executable which should help streamline ci builds by reducing linking for multipel test targets (espeically LTO processing) and by avoiding repeated transfers of common model files and test data in the case of mobile devices (and emulator) tests: gaze_add_test(). This is part of an effort to reduce build times on travis. (Using the acf package will be another step.)

* move set(DRISHTI_CI…) to top
* consolidate/remove unused test files
* update/fix extern “C” unit tests for FaceTracker
* add variables for union of all input test params in test-drishti.cpp, use extern "C" elsewhere
* remove unused cmakelists.txt for per module ut tests
* update .gitignore
* replace per test module preprocessor test directives with shared name versions: DRISHTI_DO_GPU_TESTING,
* enable DRISHTI_BUILD_C_INTERFACE=ON in development build scripts
* move sugar_include(.) up to src/CMakeLists.txt prior to  add_subdirectory(lib) to collect test sources

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ install:
   - source bin/hunter_env.sh
   
 script:
-  - bin/drishti_build.sh "${TOOLCHAIN}" "${CONFIG}" "${INSTALL}"
+  - travis_wait 200 bin/drishti_build.sh "${TOOLCHAIN}" "${CONFIG}" "${INSTALL}"
```
Comment (ruslo) [inline .travis.yml]: We will not be able to see logs but still will have timeouts...

Comment (headupinclouds) [reply .travis.yml]: `travis_wait` hides the logs?

Comment (ruslo) [reply .travis.yml]: Yes, I can't see anything: https://travis-ci.org/elucideye/drishti/jobs/289534264

Comment (headupinclouds) [reply .travis.yml]: I see.  Sample output:

```
++echo -ne 'Still running (1 of 200): bin/drishti_build.sh' gcc-5-pic-hid-sections-lto Release '--strip\r'
++sleep 60
++'[' 1 -lt 200 ']'
++count=2
++echo -ne 'Still running (2 of 200): bin/drishti_build.sh' gcc-5-pic-hid-sections-lto Release '--strip\r'
++sleep 60
++'[' 2 -lt 200 ']'
++count=3
++echo -ne 'Still running (3 of 200): bin/drishti_build.sh' gcc-5-pic-hid-sections-lto Release '--strip\r'
++sleep 60
++'[' 3 -lt 200 ']'
```

Comment (headupinclouds) [reply .travis.yml]: Does this buffer logs and dump to cout after success?

Comment (headupinclouds) [reply .travis.yml]: https://github.com/travis-ci/worker/issues/283#issue-207659669

Comment (ruslo) [reply .travis.yml]: > Does this buffer logs and dump to cout after success?

Yes, but if we hit timeout we will see nothing.

Comment (headupinclouds) [reply .travis.yml]: https://github.com/travis-ci/travis-ci/issues/7323#issuecomment-298832640

Ok.  We can remove this in the next PR.  The last android timeout was due to the global threshold I believe.  

> The job exceeded the maximum time limit for jobs, and has been terminated.

(I haven't checked recent android toolchain timeouts to see if they are due to 10 minute stdout timeout or the global timeouts.)

Comment (headupinclouds) [reply .travis.yml]: This is probably the 4th or 5th restart.  When the `travis_wait` is running it gives a direct measure of how long the build step has been running.  It seems things get to some point (like `count=38` or so) and then it just sits there as if the VM is frozen as the global clock continues:

```
++count=38
++echo -ne 'Still running (38 of 200): bin/drishti_build.sh' android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections MinSizeRel '--strip\r'
++sleep 60
```

**Comment by headupinclouds** [comment]
Merging as is until travis+android can get sorted out.

**Pull request merged**